core/trino-main/src/main/java/io/trino/type/LikeFunctions.java

```diff
@@ -245,4 +246,14 @@ private static char getEscapeChar(Slice escape)
         }
         throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "Escape string must be a single character");
     }
+
+    private static int getMatchingOffset(Matcher matcher, int at, int range)
+    {
+        try {
+            return matcher.matchInterruptible(at, range, Option.NONE);
+        }
+        catch (InterruptedException interruptedException) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Regular expression matching was interrupted, likely because it took too long. Regular expression in the worst case can have a catastrophic amount of backtracking and having exponential time complexity");
```

Comment:
```suggestion
            Thread.currentThread().interrupt();
            throw new TrinoException(GENERIC_USER_ERROR, "Regular expression matching was interrupted, likely because it took too long. Regular expression in the worst case can have a catastrophic amount of backtracking and having exponential time complexity");
```